pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egressgateway
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"strings"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/podsecuritypolicy"
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	EGWPortName               = "health"
+	DefaultEGWVxlanPort int   = 4790
+	DefaultEGWVxlanVNI  int   = 4097
+	DefaultHealthPort   int32 = 8080
+)
+
+func EgressGateway(
+	config *Config,
+) render.Component {
+	return &component{
+		config: config,
+	}
+}
+
+type component struct {
+	config *Config
+}
+
+// Config contains all the config information EgressGateway needs to render component.
+type Config struct {
+	PullSecrets  []*corev1.Secret
+	Installation *operatorv1.InstallationSpec
+	OsType       rmeta.OSType
+	EgressGW     *operatorv1.EgressGateway
+
+	egwImage          string
+	EgressGWVxlanVNI  int
+	EgressGWVxlanPort int
+	Provider          operatorv1.Provider
+}
+
+func (c *component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.config.Installation.Registry
+	path := c.config.Installation.ImagePath
+	prefix := c.config.Installation.ImagePrefix
+
+	if c.config.OsType != c.SupportedOSType() {
+		return fmt.Errorf("Egress Gateway is supported only on %s", c.SupportedOSType())
+	}
+
+	var err error
+	c.config.egwImage, err = components.GetReference(components.ComponentEgressGateway, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *component) Objects() ([]client.Object, []client.Object) {
+	objects := []client.Object{}
+	if c.config.Provider == operatorv1.ProviderRKE2 {
```

Comment:
See how this is done for example in compliance_controller.go and compliance.go. UsePSP should be passed as a boolean and is determined by the reconciler options.

For openshift, we might need security constraints.

Reply:
Have addressed UsePSP and have done it similar to compliance. Need to check about openshift.